session.go

```diff
@@ -394,39 +415,103 @@ func (s *Session) sendNoWait(hdr header) error {
 	}
 }
 
-// send is a long running goroutine that sends data
-func (s *Session) send() {
+func pooledTimer(d time.Duration) (*time.Timer, func()) {
+	t := timerPool.Get()
+	timer := t.(*time.Timer)
+	timer.Reset(d)
+	var once sync.Once
+	cancelFn := func() {
```

Comment:
couldn't think of a way to make this function idempotent and keeping thread-safety other than using `sync.Once`. The challenge is how to guard putting the timer back in the pool happens only once.